File: backend/app/schemas/notification.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel


class NotificationResponse(BaseModel):
    id: int
    user_id: int
    type: str
    title: str
    body: Optional[str] = None
    payload: dict[str, Any] = {}
    is_read: bool = False
    read_at: Optional[datetime] = None
    created_at: Optional[datetime] = None

    model_config = {"from_attributes": True}


class NotificationListResponse(BaseModel):
    notifications: list[NotificationResponse]
    unread_count: int

File: frontend/src/types/notification.ts
export interface Notification {
  id: number
  user_id: number
  type: 'match' | 'swipe_approved' | 'swipe_rejected' | 'system'
  title: string
  body: string | null
  payload?: Record<string, any>
  is_read: boolean
  read_at: string | null
  created_at: string
  related_project_id?: number
  related_user_id?: number
}

export interface NotificationListResponse {
  notifications: Notification[]
  unread_count: number
}